lib/rubocop/cop/style/extend_self.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks that `#module_function` is used over `extend self`.
+      #
+      # @example
+      #
+      #   # EnforcedStyle: module_function
```

Comment:
These days we're trying to adopt the following style for cops with configurable styles https://github.com/bbatsov/rubocop/pull/5104/files